src/schemas/index.ts

```diff
@@ -0,0 +1,26 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+export enum SchemaVersion {
+	v1_0 = '',
+	v2_0 = '2',
+	v2_1 = '2.1',
+}
+
+export const DEFAULT_SCHEMA_VERSION = SchemaVersion.v2_1;
+
+const schemas: any = {};
+schemas[SchemaVersion.v1_0] = 'v1';
+schemas[SchemaVersion.v2_0] = 'v2.0';
+schemas[SchemaVersion.v2_1] = 'v2.1';
+
+function loadJson(p: string): any {
+	const buf = fs.readFileSync(p);
```

Comment:
The synchronous read here makes me feel a little uncomfortable, although I know that the targets are always going to be small files.

Reply:
I can load all schemas at module-level and make it seem like doing a normal `require` if you feel more comfortable this way. Dynamically loading the requested schema though avoids the penalty of loading more than needed.

Reply:
I mean for everything that we're currently using it for it should be fine, I would probably prefer it to be asynchronous, but it's not a big thing for me (in it's current state). If we were doing more complex file i/o then I'd be more inclined to go async.

Reply:
I get you, but I'm arguing this is not doing arbitrary I/O and in normal JS/CS you'd probably do `var schemaV1 = require('./schema_v1.0.json')` anyway, which is synchronous.

Reply:
Oh right I see what you're saying. Fair enough I'm not too worried either way.